sw/device/silicon_owner/tock/tests/uart/src/main.rs

```diff
@@ -0,0 +1,159 @@
+// Copyright lowRISC contributors (OpenTitan project).
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#![no_main]
+#![no_std]
+
+use core::cell::Cell;
+use core::cmp::min;
+use libtock::console::Console;
+use libtock::platform::{share, AllowRo, DefaultConfig, Subscribe, Syscalls};
+use libtock::runtime::{set_main, stack_size, TockSyscalls};
+
+set_main!(main);
+stack_size!(0x800);
+
+// Reads data from the console input. The data is read in chunks whose length is
+// given by `lens`. The newly-read data is returned. Panics upon encountering an
+// error condition.
+fn read_exact<'b>(buffer: &'b mut [u8], lens: &[usize]) -> &'b [u8] {
+    assert!(buffer.len() >= lens.iter().sum());
+    let upcall: Cell<Option<(u32, u32)>> = Cell::new(None);
+    let mut cursor = 0;
+    share::scope(|handle| {
+        TockSyscalls::subscribe::<_, _, DefaultConfig, 1, 2>(handle, &upcall).unwrap();
```

Comment:
Similarly, are there symbolic constants that can be used in the `subscribe` and `allow_...` calls? (ie: these values are defined in `libtock-rs/apis/console/src/lib.rs` (although, they don't appear to be publicly exported from that module).

Reply:
There are not existing constants that can be used. I added new constants to the test and switched the code to use them.